src/operator/do.ts

```diff
@@ -48,7 +48,7 @@ import { TeardownLogic } from '../Subscription';
  * @owner Observable
  */
 /* tslint:disable:max-line-length */
-export function _do<T>(next: (x: T) => void, error?: (e: any) => void, complete?: (this: Observable<T>) => void): Observable<T>;
+export function _do<T>(this: Observable<T>, next: (x: T) => void, error?: (e: any) => void, complete?: (this: Observable<T>) => void): Observable<T>;
```

Comment:
definitely something I missed with the script.  This needs to be changed slightly, note the additional `this: Observable<T>` .

`complete?: (this: Observable<T>) => void` should be changed back to `complete?: () => void`